sqlsynthgen/make.py

```diff
@@ -412,39 +406,45 @@ def make_src_stats(
     Returns:
         The dictionary of src-stats.
     """
-    if schema_name:
-        engine = create_engine_with_search_path(dsn, schema_name)
-    else:
-        engine = create_engine(dsn, echo=False, future=True)
-
     use_smartnoise_sql = config.get("use-smartnoise-sql", True)
+    # SmartNoiseSQL doesn't support asyncio
+    use_asyncio = not use_smartnoise_sql
+    engine = create_db_engine(dsn, schema_name=schema_name, use_asyncio=use_asyncio)
+
     if use_smartnoise_sql:
+        logging.warning(
+            "SmartNoiseSQL does not support asyncio, so make-stats queries"
+            " will be run sequentially."
+        )
         dp_config = config.get("smartnoise-sql", {})
         snsql_metadata = {"": dp_config}
 
-        def execute_query(conn: Any, stat_data: Dict[str, Any]) -> Any:
+        async def execute_query(engine: Any, stat_data: Dict[str, Any]) -> Any:
             privacy = snsql.Privacy(
                 epsilon=stat_data["epsilon"], delta=stat_data["delta"]
             )
-            reader = snsql.from_connection(
-                conn.connection,
-                engine="postgres",
-                privacy=privacy,
-                metadata=snsql_metadata,
-            )
-            private_result = reader.execute(stat_data["query"])
+            with engine.connect() as conn:
+                reader = snsql.from_connection(
+                    conn.connection,
+                    engine="postgres",
+                    privacy=privacy,
+                    metadata=snsql_metadata,
+                )
+                private_result = reader.execute(stat_data["query"])
             # The first entry in the list names the columns, skip that.
             return private_result[1:]
 
     else:
 
-        def execute_query(conn: Any, stat_data: Dict[str, Any]) -> Any:
-            result = conn.execute(text(stat_data["query"])).fetchall()
+        async def execute_query(engine: Any, stat_data: Dict[str, Any]) -> Any:
+            async with engine.connect() as conn:
+                raw_result = await conn.execute(text(stat_data["query"]))
+                result = raw_result.fetchall()
             return [list(r) for r in result]
 
-    with engine.connect() as conn:
-        src_stats = {
-            stat_data["name"]: execute_query(conn, stat_data)
-            for stat_data in config.get("src-stats", [])
-        }
+    queries = config.get("src-stats", [])
+    results = await asyncio.gather(
+        *[execute_query(engine, stat_data) for stat_data in queries]
+    )
+    src_stats = {query["name"]: result for query, result in zip(queries, results)}
```

Comment:
Took me a while to work out what `stat_data` was. Is `query_block` any more intuitive?

```suggestion
    query_blocks = config.get("src-stats", [])
    results = await asyncio.gather(
        *[execute_query(engine, query_block) for query_block in query_blocks]
    )
    src_stats = {query_block["name"]: result for query_block, result in zip(query_blocks, results)}
```